### DISCUSSION TITLE
The entire course

### DISCUSSION THREAD

**Discussion by LayZeeDK** [this-is-learning/rxjs-fundamentals-course#6] [Content contributions]
Discussing and coordinating efforts spanning [the entire course](https://github.com/this-is-learning/rxjs-fundamentals-course).

**Comment by LayZeeDK** [comment]
# Course title
Please suggest and vote for the best course title in this comment thread.

**Reply by LayZeeDK** [reply]
Suggestion: RxJS Fundamentals

**Reply by LayZeeDK** [reply]
Suggestion: Fundamentals of RxJS

**Reply by LayZeeDK** [reply]
Suggestion The Fundamentals of RxJS

**Reply by LayZeeDK** [reply]
Suggestion: RxJS Foundation

**Comment by LayZeeDK** [comment]
# Course content
Currently, 5 parts are planned and they all have content to some extent:
1. [Async programming issues and RxJS. Promises vs Observables](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/1)
1. [Reactive Programming with RxJS](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/2)
1. [The most common RxJS operators](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/3)
1. [How RxJS is used by Angular](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/4)
1. [Everything is a stream: Pushed-based architecture](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/5)

**Reply by LayZeeDK** [reply]
@nartc is looking to add existing Vietnamese content and translate it to English, then figure out how it could best fit into this course.

**Reply by Andrei0872** [reply]
Maybe it would become a bit overwhelming, but since in RxJS is inevitable to have to deal with _observables of observables_, maybe it's worth mentioning them and how they work(at a high level).

**Reply by LayZeeDK** [reply]
Are you thinking about something like this example from the NgRx Effects docs?
```ts
import { Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';
import { EMPTY } from 'rxjs';
import { map, mergeMap, catchError } from 'rxjs/operators';
import { MoviesService } from './movies.service';

@Injectable()
export class MovieEffects {

  loadMovies$ = createEffect(() => this.actions$.pipe(
    ofType('[Movies Page] Load Movies'),
    mergeMap(() => this.moviesService.getAll()
      .pipe(
        map(movies => ({ type: '[Movies API] Movies Loaded Success', payload: movies })),
        catchError(() => EMPTY)
      ))
    )
  );

  constructor(
    private actions$: Actions,
    private moviesService: MoviesService
  ) {}
}
```

**Reply by Andrei0872** [reply]
I haven't thought of anything specifically, but the example you provided is very pertinent. Maybe we could highlight the use cases of observables of observables, like the one from above: consuming data from other observables without having to deal with the subscriptions yourself. 

Other examples might be the well-known autocomplete feature, where `switchMap` helps a lot, or when you want to ignore events while an action is doing its job(`exhaustMap`).

**Reply by Armenvardanyan95** [reply]
I think I can also add a separate section on higher order Observables as part 6. What do you think, should I get started?

**Reply by LayZeeDK** [reply]
What do you all think, is this an entry-level topic?

**Reply by Andrei0872** [reply]
IMO, it is something that should be at least mentioned.

**Reply by kosich** [reply]
I agree, an autocomplete example seems to be canonical and (imho) should be definitely described. Maybe, not deep enough to dive into *Map difference and subscriptions/unsubscriptions (this probably belongs to the next course).

Also, will there be a part about React (and maybe others) with RxJS? 🙂

**Reply by LayZeeDK** [reply]
Hi @kosich,

Thanks for joining our discussion. [Part 5](https://github.com/this-is-learning/rxjs-fundamentals-course/blob/main/part-5/part-5.md) includes a simple example of reusing Akita services between a React and an Angular application. Akita is based on RxJS observables.

**Reply by LayZeeDK** [reply]
@Armenvardanyan95 as flattening operators are already mentioned in [Part 1](https://github.com/this-is-learning/rxjs-fundamentals-course/blob/main/part-1/part-1.md), I suggest we improve that section for now.

**Reply by LayZeeDK** [reply]
@kosich [Part 2](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/2) was supposed to have a section on subscriptions.

**Comment by LayZeeDK** [comment]
# Repository access
Reply here to get an invite for the @this-is-learning/courses team which grants you write access to this repository.

**Reply by rinckd** [reply]
I would love to join this group. I would like to focus on Part 5. Pushed based architecture.

**Reply by LayZeeDK** [reply]
I sent you an invite to the @this-is-learning/courses team 📩

**Comment by LayZeeDK** [comment]
# Discord server
We also discuss our efforts in the private This is Learning Discord server. Reply here to request a join link.

**Reply by rinckd** [reply]
Would love to join the discord server as well.

**Reply by LayZeeDK** [reply]
Check you Twitter inbox, @rinckd 📩
